torch/lib/c10d/logger.cpp

```diff
@@ -65,9 +65,14 @@ void Logger::set_env_variables() {
 void Logger::set_parameter_stats() {
   ddp_logging_data_->num_parameter_tensors = reducer_->replicas_[0].size();
   ddp_logging_data_->total_parameter_size_bytes = 0;
-  for (const auto& t : reducer_->replicas_[0]) {
+  std::set<std::string> unique_dtypes;
+  for (auto t : reducer_->replicas_[0]) {
     ddp_logging_data_->total_parameter_size_bytes +=
         t.numel() * t.element_size();
+    unique_dtypes.insert(std::string(t.dtype().name()));
```

Comment:
Could we remove `unique_dtypes` and just do `ddp_logging_data_->dtypes.push_Back(std::string(t.dtype().name())` directly in this loop?